src/main/java/net/minecraftforge/common/util/BlockSnapshot.java

```diff
@@ -309,25 +273,30 @@ public boolean equals(Object obj)
     public int hashCode()
     {
         int hash = 7;
-        hash = 73 * hash + this.getPos().getX();
-        hash = 73 * hash + this.getPos().getY();
-        hash = 73 * hash + this.getPos().getZ();
         hash = 73 * hash + this.getMeta();
         hash = 73 * hash + this.getDimId();
+        hash = 73 * hash + this.getPos().hashCode();
```

Comment:
hashCode is not required to be consistent even between JVM runs. Relying on it is a bug.